js/render/draw_background.js

```diff
@@ -6,14 +6,16 @@ module.exports = drawBackground;
 
 function drawBackground(painter, sourceCache, layer) {
     const gl = painter.gl;
-    const transform = painter.transform;
-    const tileSize = transform.tileSize;
     const color = layer.paint['background-color'];
     const image = layer.paint['background-pattern'];
     const opacity = layer.paint['background-opacity'];
 
     if (painter.isOpaquePass !== (!image && color[3] === 1)) return;
```

Comment:
Whoa, you're right! Fixing this also fixes https://github.com/mapbox/mapbox-gl-js/issues/2925 🎉